### PR TITLE
:recycle: refactor: Support disable clientFetch by default

### LINKED ISSUE #3108: [Bug] qwen的api key在这里用不了
### 📦 部署环境

Docker

### 📌 软件版本

lastest

### 💻 系统环境

Ubuntu

### 🌐 浏览器

Edge

### 🐛 问题描述

🤯 随便聊聊 · LobeChat 和另外 14 个页面 - 个人 - Microsoft​ Edge 2024_7_1 23_58_52
报的错qwen的文档里没有，另外同一个key在chatbox上可以用
Chatbox 2024_7_1 23_48_21

### 📷 复现步骤

🤯 随便聊聊 · LobeChat 和另外 14 个页面 - 个人 - Microsoft​ Edge 2024_7_2 0_03_57

### 🚦 期望结果

搞不懂这是不是bug

### 📝 补充信息

_No response_

### PR DESCRIPTION
**Pull request by cy948**
#### 💻 变更类型 | Change Type

- [ ] ✨ feat
- [ ] 🐛 fix
- [x] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] ⚡️ perf
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

- ♻️ `src/config/modelProviders/index.ts`: 从 `modelConfig` 中判断是否允许“客户端请求”
- ♻️ `src/config/modelProviders/qwen.ts`: 默认禁用 `qwen` provider 的“客户端请求”；
- ♻️ `src/store/user/slices/modelList/selectors/modelConfig.ts`: 若modelProvider中禁用了provider的“客户端请求”，则不启用；
- 🧪 `src/store/user/slices/modelList/selectors/modelConfig.test.ts`: 测试 Qwen Provider 能否被拦截，即使其符合其它启用“客户端请求”的条件；

#### 📝 补充信息 | Additional Information
- Preview on [Vercel](https://lobe-chat-git-refactor-clientfetch-disabled-cy948s-projects.vercel.app/settings/llm)

-  fix #3108

### CODE REVIEW

File: src/app/(main)/settings/llm/components/ProviderConfig/index.tsx [hunk 1]
```diff
@@ -191,11 +193,13 @@ const ProviderConfig = memo<ProviderConfigProps>(
        * Conditions to show Client Fetch Switch
        * 1. provider is not disabled browser request
        * 2. provider show browser request by default
-       * 3. Provider allow to edit endpoint and the value of endpoint is not empty
-       * 4. There is an apikey provided by user
+       * 3. provider is not disable fetch on client by default
+       * 4. Provider allow to edit endpoint and the value of endpoint is not empty
+       * 5. There is an apikey provided by user
        */
       !disableBrowserRequest &&
```
Comment (cy948) [inline src/app/(main)/settings/llm/components/ProviderConfig/index.tsx]: 禁用“client fetch”的功能似乎与此处重合。

File: src/const/settings/llm.ts
Comment (arvinxx) [inline src/const/settings/llm.ts:1]: 不应该写在 settings 里，应该放在 model provider 的config 比较合适。settings 的内容是可以被覆盖的。

Comment (cy948) [reply src/const/settings/llm.ts:1]: 已转移至 model config

File: src/config/modelProviders/qwen.ts [hunk 2]
```diff
@@ -50,6 +52,8 @@ const Qwen: ModelProviderCard = {
     },
   ],
   checkModel: 'qwen-turbo',
+  defaultShowBrowserRequest: false,
+  disableBrowserRequest: true,
```
Comment (arvinxx) [inline src/config/modelProviders/qwen.ts:55]: 配置了下面这个，应该就不用配置上面的了

File: src/store/user/slices/modelList/selectors/modelConfig.ts [hunk 3]
```diff
@@ -21,6 +22,8 @@ const isProviderFetchOnClient = (provider: GlobalLLMProviderKey | string) => (s:
   // If the provider in the whitelist, follow the user settings
   if (providerWhitelist.has(provider) && typeof config?.fetchOnClient !== 'undefined')
     return config?.fetchOnClient;
+  // If the provider already disable broswer request in model config, force on Server.
+  if (isProviderDisableBroswerRequest(provider)) return false;
```
Comment (arvinxx) [inline src/store/user/slices/modelList/selectors/modelConfig.ts]: 这一行应该最先判断吧？然后再是有没有用户配置

File: src/config/modelProviders/qwen.ts (hunk 2 above)
Comment (cy948) [reply src/config/modelProviders/qwen.ts:55]: 已去掉

File: src/store/user/slices/modelList/selectors/modelConfig.ts (hunk 3 above)
Comment (cy948) [reply src/store/user/slices/modelList/selectors/modelConfig.ts]: 判断放到最前面了

**Comment by cy948** [comment]
@JonBCraig We diasble the options for Qwen cause it has cross-origin problem before. 
- But you can add a custom OpenAI provider to use Qwen with client request options to test if it work now. If worked, it's wellcome to make pull request to change the default settings. 
- Moreover, the desktop version might well fit your use case which request with client but store data in your Vercel self-hosted instance.

**Pull request merged**